src/mmw/apps/modeling/mapshed/calcs.py

```diff
@@ -326,3 +326,71 @@ def weather_data(ws, begyear, endyear):
                 prcps[year, month, day] = p if p != WEATHER_NULL else None
 
     return temps, prcps
+
+
+def curve_number(n_count, ng_count):
+    """
+    Given a dictionary mapping NLCD codes to counts of cells, and another
+    mapping pairs of NLCD codes and Hydrological Soil Groups, returns the
+    curve number for each non-urban MapShed land use type by calculating the
+    average hydrological soil group for each land use type, rounded to the
+    nearest integer, and looking up the value in the CURVE_NUMBER table.
+    """
+
+    # Calculate average hydrological soil group for each NLCD type by
+    # reducing [(n, g): c] to [n: avg(g * c)]
+    n_gavg = {}
+    for (n, g), count in ng_count.iteritems():
+        n_gavg[n] = (float(g) * count / n_count[n] + n_gavg.get(n, 0))
```

Comment:
Consider removing the outer parens, which don't say much, and use parens to group the subexpressions.